nondex-test/src/test/java/edu/illinois/nondex/core/AbstractCollectionTest.java

```diff
@@ -56,16 +58,25 @@ protected void assertParameterized(T ds, Object derived, String str) {
         }
     }
 
-    protected void assertEqualstUnordered(String msg, String expected, String actual) {
-        Assert.assertEquals(msg + ": " + expected + " =/= " + actual, expected.length(), actual.length());
-        String trimmed = expected.substring(1, expected.length() - 1);
+    private String[] formatString(String msg) {
+        String trimmed = msg.substring(1, msg.length() - 1);
         String[] elems = trimmed.split(",");
-        // TODO(gyori): fix and make this more robust. It does not check duplicates, substrings, etc.
         for (int i = 0; i < elems.length; i++) {
             elems[i] = elems[i].trim();
-            Assert.assertTrue(msg + ": " + trimmed + " =/= " + actual, actual.contains(elems[i]));
         }
+        return elems;
+    }
 
+    protected void assertEqualstUnordered(String msg, String expected, String actual) {
+        String[] actualTokenized = this.formatString(actual);
+        String[] expectedTokenized = this.formatString(expected);
 
+        Arrays.sort(actualTokenized);
+        Arrays.sort(expectedTokenized);
+        if (Arrays.equals(actualTokenized, expectedTokenized)) {
+            Assert.assertTrue(msg + ": " + expectedTokenized + "=/= " + actualTokenized, true);
```

Comment:
`assertTrue(message, true)` can never fail. What is the goal of this assert?

Reply:
It seems you want to use `assertArraysEquals`.

Reply:
sure. we will incorportate this